src/uwtools/config.py

```diff
@@ -752,3 +753,22 @@ def create_config_obj(
                     raise ValueError(err_msg)
                 # Dump to file:
                 dump_method(path=outfile, cfg=config_obj)
+
+
+def print_config_section(config: dict, section_path: List[str], log: Logger) -> None:
+    """
+    Descends into the config via the given section keys, then prints the contents of the located
+    subtree as key=value pairs, one per line.
+    """
+    for key in section_path:
+        if isinstance(config[key], dict):
+            config = config[key]
```

Comment:
What happens here if `key` is not in `config`? Should we handle that explicitly, too?

Reply:
Would something like
```
keys = []
for key in section_path:
    keys.append(key)
    current_path = " -> ".join(keys)
    try:
        subconfig = config[key]
    except KeyError:
        msg = f"Bad config path: {current_path}"
        log.error(msg)
        raise UWConfigError(msg)
    if isinstance(subconfig, dict):
        config = subconfig
    else:
        msg = f"Value at {current_path} must be a dictionary"
        log.error(msg)
        raise UWConfigErrror(msg)
```
give the error check @christinaholtNOAA suggests, plus, usefully, the actual path from the top of the config down to the place we found an error? Maybe something similar below in the value-type check. (Also `msg` / `log` / `raise` starts to look like a pattern...)